File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>result(nums.size(),1);
        int acc = 1; 
        for(int i = 0; i< nums.size()-1; i++)
        {
            result[i+1] = acc*nums[i];
            acc*=nums[i];
        }
        acc =1 ; 
        for(int i = nums.size()-1; i >= 1; i--)
        {
            result[i-1] *= (acc*nums[i]);
            acc*=nums[i];
        }
        return result;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let result = [];
        for(let i = 0; i< nums.length; i++)
            result.push(1);
        let acc = 1; 
        for(let i = 0; i< nums.length-1; i++)
        {
            result[i+1] = acc*nums[i];
            acc*=nums[i];
        }
        acc =1 ; 
        for(let i = nums.length-1; i >= 1; i--)
        {
            result[i-1] *= (acc*nums[i]);
            acc*=nums[i];
        }
        return result;
    
};